### PR TITLE
add PayPal php libraries

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

https://developer.paypal.com/docs/checkout/reference/server-integration/setup-sdk/
Slack discussion: https://givewp.slack.com/archives/C0FAGC83C/p1593709856067600

PayPal provided PHP SDK to perform PayPal checkout rest API queries. This PR adds PayPal libraries and the libraries are scoped in `Give/Vendors/PayPal` to prevent future conflict.

## Affects

This does not impact the existing functionality but adds a library for future use.

## What to test

- [ ] do you see PayPal library in `Give\Vendors` when running `composer install`?

## Screenshots:

image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -7,14 +7,17 @@
...
     },
     "require-dev": {
         "squizlabs/php_codesniffer": "*",
         "dealerdirect/phpcodesniffer-composer-installer": "^0.6",
         "phpcompatibility/php-compatibility": "^9.3",
         "phpcompatibility/phpcompatibility-wp": "*",
-        "wp-coding-standards/wpcs": "*"
+        "wp-coding-standards/wpcs": "*",
+        "jetbrains/phpstorm-stubs": "dev-master",
+        "humbug/php-scoper": "dev-master"
```
Comment (JasonTheAdams) [inline composer.json:20]: Please don't use "dev-master" unless absolutely necessary. We can use "^0.13" instead.

File: config/php-scoper/paypal.inc.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
+declare( strict_types=1 );
+
+use Isolated\Symfony\Component\Finder\Finder;
+
+return [
+	'prefix'   => 'Give\Vendor\PayPal',
```
Comment (JasonTheAdams) [inline config/php-scoper/paypal.inc.php:7]: What's the resulting namespace, here? It seems like the last "\PayPal" part shouldn't be necessary.

Comment (ravinderk) [reply config/php-scoper/paypal.inc.php]: Fixed

File: composer.json (hunk 1 above)
Comment (ravinderk) [reply composer.json]: fixed

File: composer.json [hunk 3]
```diff
@@ -27,13 +30,21 @@
     "scripts": {
         "post-install-cmd": [
             "\"vendor/bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs || true",
-            "@install-codestandards"
+            "@install-codestandards",
+            "composer prefix-dependencies"
```
Comment (JasonTheAdams) [inline composer.json:34]:
```suggestion
            "@prefix-dependencies"
```

**Comment by JasonTheAdams** [comment]
@ravinderk Ok... so I spent a good hour or two seeing if I could solve the riddle of package prefixing. Turns out this is a really complicated problem and there are multiple folks trying to solve it. In addition to PHP Scoper, I tried [mozart](https://github.com/coenjacobs/mozart), and [imposter](https://github.com/TypistTech/imposter). Turns out that PHP Scoper is the most mature of the libraries.

What I found was that the way PHP Scoper is designed to be used is to actually go through the vendor _and_ src classes, and update the vendor namespaces and usage. That said, I looked some examples such as the [Delicious Brains article](https://deliciousbrains.com/php-scoper-namespace-composer-depencies/), and found that this isn't really how many folks are using it.

There are a number of caveats and challenges with the library, but if we're very specific in how we use it then it seems like it may do the job. I feel bad saying this, but frankly your initial direction seems to be the best way I can think to go. The problem with the `vendors-prefixed` idea, is that this only really works if you let PHP Scoper do the whole project, in which case it generates a `scoper-autoload.php` directory. I really wish I could get a peek as to how DB is using it, as they seem to _not_ worry about it locally and only use it during their final build process.

I also tried _replacing_ the packages in the `vendor` directory and I feel like I got close, but I'm not sure if it's the final solution.

Finally, I found myself in a really weird situation. If I leave "humbug/php-scoper" in the composer.json file, then I get on error on every admin screen (see below). If I remove that package, _delete the composer.lock file_, and then install again, then the error goes away.

Syntax Error

This has turned into a royal pain. I'm seriously tempted to just add the packages normally and move forward, at least for now.

**Comment by ravinderk** [comment]
@JasonTheAdams I am closing this pr for now but I will keep this branch to continue in same branch

**Comment by JasonTheAdams** [comment]
@ravinderk I see that you closed this PR and then just added the libraries to the epic branch. Can you please help me understand _why_ you went that route?

**Comment by JasonTheAdams** [comment]
For future reference, this will be _very_ useful: https://developer.yoast.com/blog/safely-using-php-dependencies-in-the-wordpress-ecosystem/

**Comment by ravinderk** [comment]
@JasonTheAdams I thought you ask me to do that in a comment. I do not want to lose work in this branch, so I close this pr without deleting the existing branch.

> I'm seriously tempted to just add the packages normally and move forward, at least for now.

**Pull request closed without merging**